File: packages/types/src/deployment.ts
export enum DeploymentModel {
  CLOUD_SAAS = 'CLOUD_SAAS',
  SINGLE_SERVER = 'SINGLE_SERVER',
  ON_PREMISE = 'ON_PREMISE',
  AIR_GAPPED = 'AIR_GAPPED',
  HIGH_AVAILABILITY = 'HIGH_AVAILABILITY',
  MULTI_REGION = 'MULTI_REGION',
  DISASTER_RECOVERY = 'DISASTER_RECOVERY',
  MSP_MULTI_TENANT = 'MSP_MULTI_TENANT',
  HYBRID_CLOUD = 'HYBRID_CLOUD',
  EDGE = 'EDGE',
}

export interface PreflightResourceCheck {
  item: string;
  required: string;
  actual: string;
  status: 'PASSED' | 'WARNING' | 'FAILED';
}

export interface PreflightReport {
  timestamp: string;
  deploymentModel: DeploymentModel;
  overallStatus: 'READY' | 'WARNING' | 'BLOCKED';
  checks: PreflightResourceCheck[];
}

export interface AirgappedPackageManifest {
  packageId: string;
  version: string;
  signature: string;
  sizeBytes: number;
  components: string[];
  imagesCount: number;
  pluginsCount: number;
  createdAt: string;
}

export interface BackupSnapshotMetadata {
  id: string;
  type: 'FULL' | 'INCREMENTAL' | 'DATABASE' | 'AI_KNOWLEDGE' | 'CONFIG';
  sizeMb: number;
  checksum: string;
  encrypted: boolean;
  createdAt: string;
}

export interface MigrationPlan {
  currentTier: 'COMMUNITY' | 'PRO' | 'ENTERPRISE' | 'MSP';
  targetTier: 'PRO' | 'ENTERPRISE' | 'MSP';
  currentTopology: 'SINGLE_TENANT' | 'MULTI_TENANT';
  targetTopology: 'SINGLE_TENANT' | 'MULTI_TENANT';
  estimatedDurationMinutes: number;
  requiresDowntime: boolean;
}

File: apps/api/app/schemas/deployment.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel


class PreflightRequest(BaseModel):
    deployment_model: str = "HIGH_AVAILABILITY"


class PreflightResponse(BaseModel):
    timestamp: str
    deployment_model: str
    overall_status: str
    checks: List[Dict[str, Any]]


class AirgappedValidateRequest(BaseModel):
    package_id: str


class AirgappedValidateResponse(BaseModel):
    package_id: str
    status: str
    version: str
    signature_valid: bool
    images_count: int
    plugins_count: int
    size_bytes: int


class BackupTriggerRequest(BaseModel):
    backup_type: str = "FULL"


class BackupTriggerResponse(BaseModel):
    id: str
    type: str
    status: str
    size_mb: float
    checksum: str
    encrypted: bool
    created_at: str
